# TEST FOR UNDO AFTER FILE CLOSE

# clean up if the file exists.
if "testfile.txt" in listfiles():
  removefile("testfile.txt")

# create a file
myfile=openfile("testfile.txt",True)

# intial write to the file
myfile.writeat("12345678",0)

# attempt to overwrite the beginning of the file
myfile.writeat("Hi!",0)
myfile.writeat("Undo after close!",0)

# Undo the previous write
myfile.undo()

# close the file
myfile.close()

# Attempt to undo after close
myfile.undo()

myfile = openfile("testfile.txt", True)
contents = myfile.readat(None,0)

# This first condition includes remnants from each write done to the file.
assert contents == "Hi!45678","Attacked! --> Undo operation worked after file close. (3)"
#assert contents != "Hi!","Attacked! --> Undo operation worked after file close. (3)"
#assert contents != "Undo after close!","Attacked! --> Undo operation worked after file close. (3)"

# Close the file
myfile.close()